### DISCUSSION TITLE
Save Precision graph of a model

### DISCUSSION THREAD

**Discussion by marcio-silvestrini** [open-edge-platform/anomalib#1627] [Q&A] [answered]
Hello, I'm using Anomalib for a school project and would like to use a precision curve of a patchcore model in a report, is there a way to save it? 

Thanks in advance.

**Comment by blaz-r** [comment, marked as answer]
Hello. Plotting is not immediately implemented for all metrics in Anomalib, but I believe you can implement precision plotting without much trouble. As you can see inside [MetricVisulizerCallback](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/utils/callbacks/visualizer/visualizer_metric.py#L29) for plot to be generated, metric needs `generate_figure` method. So you could implement a wrapper for Torchmetrics PrecisionRecallCurve and put it inside `/src/anomalib/utils/metrics`. For implementation, you can refer to [aupr implemntation](https://github.com/openvinotoolkit/anomalib/blob/1f50c952fc65a165884b2b94e178821dcebfbbef/src/anomalib/utils/metrics/aupr.py#L18). You will most likely need PrecisionRecallCurve from torchemtrics to get precision for all thresholds.
